app/javascript/components/namespace-form/index.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { API } from '../../http_api';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import schema from './namespace-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import handleFailure from '../../helpers/handle-failure';
+
+const getInitialValues = (isDomain, namespaceId, f) => {
+  miqSparkleOn();
+  API.get(`/api/automate_${isDomain ? 'domains' : 'namespaces'}/${namespaceId}?attributes=name,description,enabled,source`).then((result) => {
+    miqSparkleOff();
+    f.setInitialValues({ name: result.name, description: result.description, enabled: result.enabled });
+    // FIXME Retrieve readability from the API directly instead of the source (property is missing from the API)
+    f.setDisabled(!['user', 'user_locked'].includes(result.source));
+  });
+};
+
+const handleSubmit = (isDomain, namespaceId, parentId, values, redirectUrl) => {
```

Comment:
You can move getInitialValues and handleSubmit to the component, then you don't need to pass so many arguments

Reply:
It has some drawbacks on performance, and I think it is better to have them outside (5 is a bearable amount of param I think).

Reply:
I don't believe the drawback would be noticeable, but okay, makes sense.